### ISSUE TITLE
Accuracy of validation data in text classification

### ISSUE THREAD

**Issue by FinScience** [rstudio/keras#314]
I am trying a text classification model using Keras in R. I am facing trouble in improving my model accuracy on validation data.

Any help would be great. Sharing the cleaned file here along with the code.

[https://drive.google.com/open?id=1g8Cmw2bmAI9DnOU-rB4sjsOeBuFp6NUy](url)

````
#Normalize data

data[,1:(ncol(data)-1)] = normalize(data[,1:(ncol(data)-1)])

data[,ncol(data)] = as.numeric(data[,ncol(data)]) - 1

set.seed(128)

ind = sample(2,nrow(data),replace = T,prob = c(0.7,0.3))

training = data[ind==1,1:(ncol(data)-1)]

test = data[ind==2,1:(ncol(data)-1)]

traintarget = data[ind==1,ncol(data)]

testtarget = data[ind==2,ncol(data)]
# One hot encoding
trainLabels = to_categorical(traintarget)
testLabels = to_categorical(testtarget)
print(testLabels)

model = keras_model_sequential()

model %>% 
  layer_dense(units = 150, activation = 'relu', input_shape = c(520)) %>% 
  layer_dense(units = 50, activation = 'relu') %>%
  layer_dense(units = 9, activation = 'softmax') 


model %>%
  compile(loss = 'categorical_crossentropy', optimizer = 'adam',metrics = 'accuracy')


history = model %>%
  fit(training,
      trainLabels,
      epoch = 300,
      batch_size = 32,
      validation_split = 0.2)

prob = model %>%
  predict_proba(test)

pred = model %>%
  predict_classes(test)

table2 = table(Predicted = pred, Actual = testtarget)

cbind(prob,pred,testtarget)
```

`Epoch 299/300
295/295 [==============================] - 0s 653us/step - loss: 0.3872 - acc: 0.7898 - val_loss: 5.1880 - val_acc: 0.1757
Epoch 300/300
295/295 [==============================] - 0s 653us/step - loss: 0.3803 - acc: 0.8102 - val_loss: 5.4205 - val_acc: 0.1351`

**Comment by Mrugankakarte** [comment]
Have you tried LSTM or GRU model? Try Recurrent neural networks. Check this out [link](http://www.deeplearningbook.org/)
